core/tee/tee_svc_storage.c

```diff
@@ -455,22 +456,36 @@ TEE_Result tee_svc_storage_obj_open(uint32_t storage_id, void *object_id,
 
 	res = tee_svc_storage_read_head(sess, o);
 	if (res != TEE_SUCCESS) {
+		if (res == TEE_ERROR_CORRUPT_OBJECT) {
```

Comment:
@jamescfkung I agree with you it is a a kind of redundant but I would prefer to use this function for simplicity and consistency. What do you think ?